please.sh

```diff
@@ -3462,8 +3463,13 @@ mention () { # <what, e.g. bug-fix, new-feature> <release-notes-item>
 
 			:2;s/^/### '"$what"'\n\n'"$quoted"'\n\n/;b5;
 
-			:3;/^\*/b4;n;b3;:4;n;/^\*/b4;
-			s/^/'"$quoted"'\n/;b5;
+			:3;/^\*/b4;n;b3;
+			
+			:4;/.*'"$search"'.*/b6;n;/^\*/b4;
+
+            :7;s/^/'"$quoted"'\n/;b5;
+
+			:6;N;s/.*\n//;/^\*/b4;b7;
```

Comment:
Hehe... it is tempting to extend the sed script, but it does look more convoluted yet than before. I guess I regret my choices, and I should just have gone with a node.js or Perl snippet here?

If you move this *before* the `:7` line, you can avoid the `b7` altogether (and hence also the label `:7`). Then also that `b5` in the line starting with `:7` would become unnecessary again (I just realized that I left that in by mistake).

The `s/.*\n//` is correct, though: the `.` does not match a newline, so this really only deletes the (just added) first line of the buffer. To make this a bit more explicit, the anchor `^` could be used.

Oh, and could you adjust the labels so that they are in increasing order again?

From what I can tell, this code works correctly, though.

Reply:
Trying to be a bit more clear here: to me, the `:6` seems like an `else` clause. So I could see it being the `/<delete-condition>/b5;n;b6:5;N;:6` dance below (where `b5` was renamed to `b7`):

```
:1;n;
/^### '"$what"'/b3;
/^### Bug Fixes/b2;
/^## Changes since/b2;
b1;

:2;s/^/### '"$what"'\n\n'"$quoted"'\n\n/;b7;
:3;/^\*/b4;n;b3;

:4;/.*'"$search"'.*/b5;n;b6:5;N;s/^.*\n//:6;/^\*/b4;
s/^/'"$quoted"'\n/;
:7;n;b7
```

There is still one caveat, in that the `search` variable might contain too ambiguous a needle, and that it will be interpreted as regex, so special characters should probably be escaped by a backslash.